src/models/voting.ts

```diff
@@ -0,0 +1,30 @@
+import mongoose from 'mongoose';
+import { VOTING_STAGE } from '../types/voting';
+import { getStringEnumValues } from '../utils/enum';
+
+const VotingSchema = new mongoose.Schema({
+  stage: {
+    type: String,
+    enum: getStringEnumValues(VOTING_STAGE),
+    required: true,
+  },
+});
+
+VotingSchema.statics.getVotingDoc = async () => {
+  let votingDoc = await VotingModel.findOne({});
+  if (!votingDoc) {
+    votingDoc = await VotingModel.create({ stage: VOTING_STAGE.Closed });
+  }
+  return votingDoc;
+};
```

Comment:
I'm not too huge of a fan of this. It seems to me like `!votingDoc` will only be true once and false every other time. This code makes me fee like for some reason `votingDoc` will not be available sometimes and we'll have to make a new one. Since we (the devs) are the only ones who have to deal with the case where the voting doc doesn't exist, maybe we can write a dev route that creates the voting doc if it doesn't already exist for the devs to use. Rather than have that additional check that is always gonna be false on each call.

What do you think?

Reply:
I don't think the additional check is a big problem. We are checking if the doc exists each time even though only it will only not exist on the first run on a fresh DB but in terms of performance, it shouldn't be any different at all. And this way we don't have to run an extra command or hit a dev route before we can get up and running with the app.

If it's about clarity how about if I changed the name to `getOrCreateVotingDoc`?

Reply:
It's actually fine😅 Just feels a lil weird but it's probably okay
If anything you can probably rename to `getDoc` because they'll only ever be 1 and we have to call it on the voting model anyway.

Reply:
Sounds good to me! We can even get rid of the Model completely and just export the single doc but I'll just update the name for now

Reply:
Sounds good to me! We can even get rid of the Model completely and just export the single doc but I'll just update the name for now